crates/matrix-sdk-crypto/src/olm/group_sessions/inbound.rs

```diff
@@ -217,6 +235,16 @@ impl InboundGroupSession {
         &self.signing_keys
     }
 
+    /// If from a trusted source will keep the safe flag
+    /// If not will override and set to unsafe.
+    /// This takes ownership of the old version
+    pub fn from_trusted_source(self, trusted: bool) -> InboundGroupSession {
```

Comment:
If you take my advice from above, then this method isn't needed either. It's also quite scary to have such a method where we can just forget to call it and suddenly things get trusted that shouldn't be.

Reply:
yes, I wanted to force call it in the try_from(), but it can't have a parameter

Reply:
I introduced a `ClaimedInboundGroupSession` struct.
This can be constructed from KeyExport or KeyForwards
It can be transform back to an `InboundGroupSession` by using `to_group_session` that take in parameter whether the source is trusted or not. If source is not trusted the `claimed_trust` is overidden to false